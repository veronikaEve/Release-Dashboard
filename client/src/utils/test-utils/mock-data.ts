import { VersionData } from "../../@types";

export const mockVersionData: VersionData[] = [
  {
    associatedUser: "@johnSmith",
    date: "11/03/2022",
    versionFrom: "v7.8.2",
    versionTo: "v7.9.6",
    _id: "63122a332e9904df79c834ff",
  },
  {
    associatedUser: "@billy1234",
    date: "09/05/2022",
    versionFrom: "v7.7.10",
    versionTo: "v7.8.2",
    _id: "6313b230e6c802832ffeee01",
  },
];
