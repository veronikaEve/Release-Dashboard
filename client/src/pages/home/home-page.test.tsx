import { render, screen } from "@testing-library/react";
import React from "react";

import HomePage from "./home-page";
import { mockVersionData } from "../../utils/test-utils/mock-data";

describe("HomePage", () => {
  const props = {
    versionData: mockVersionData,
  };

  test("should match snapshot", () => {
    const { asFragment } = render(<HomePage {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should have the version table", () => {
    render(<HomePage {...props} />);
    const versionTable = screen.getByTestId("version-table");
    expect(versionTable).toBeInTheDocument();
  });
});
