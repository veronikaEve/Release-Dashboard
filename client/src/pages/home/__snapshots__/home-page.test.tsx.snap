// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-5"
    style="display: grid; gap: 40px;"
  >
    <div
      class="table-wrapper"
    >
      <div
        class="table-wrapper__scroll"
      >
        <table
          class="version-table table table-sm table-bordered table-hover"
          data-testid="version-table"
        >
          <thead
            class="version-table__header"
          >
            <tr>
              <th>
                Date
              </th>
              <th>
                From
              </th>
              <th>
                To
              </th>
              <th>
                Excuted by
              </th>
            </tr>
          </thead>
          <tbody
            class="version-table__body"
            data-testid="version-table__body"
          >
            <tr
              class=""
              data-testid="table-row-0"
            >
              <td>
                11/03/2022
              </td>
              <td>
                v7.8.2
              </td>
              <td>
                v7.9.6
              </td>
              <td>
                @johnSmith
              </td>
            </tr>
            <tr
              class=""
              data-testid="table-row-1"
            >
              <td>
                09/05/2022
              </td>
              <td>
                v7.7.10
              </td>
              <td>
                v7.8.2
              </td>
              <td>
                @billy1234
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="release-info-card text-center card"
    >
      <div
        class="card-header"
      >
        Release Info
      </div>
      <div
        class="card-body"
      >
        <div
          class="info-group row"
        >
          <div
            class="info-group__label col"
          >
            Branch name
          </div>
          <div
            class="info-group__info col"
          >
            wha
          </div>
        </div>
        <div
          class="info-group row"
        >
          <div
            class="info-group__label col"
          >
            Current version
          </div>
          <div
            class="info-group__info col"
          >
            ehhh
          </div>
        </div>
        <div
          class="info-group row"
        >
          <div
            class="info-group__label col"
          >
            Commit hash
          </div>
          <div
            class="info-group__info col"
          >
            grsfe
          </div>
        </div>
        <div
          class="info-group row"
        >
          <div
            class="info-group__label col"
          >
            Last Update
          </div>
          <div
            class="info-group__info col"
          >
            fvr
          </div>
        </div>
        <div
          class="info-group row"
        >
          <div
            class="info-group__label col"
          >
            Opened By
          </div>
          <div
            class="info-group__info col"
          >
            cswwww
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
